Expose transaction setting

alcaeus opened the pull request:
This complements https://github.com/doctrine/mongodb-odm/pull/2586 by exposing the `use_transactional_flush` setting to the bundle configuration.

alcaeus committed:
Expose use_transactional_flush setting

malarzm reviewed (commented):
Configuration reference from docs should be updated in same PR :)

alcaeus committed:
Add transaction setting to config reference

alcaeus committed:
Support versions of ODM without transaction support

alcaeus commented:
Updated to target 5.0.x, which we may bump to 5.1.x. I've also added logic to support older ODM versions that don't support transactional flushes. We can decide whether to bump the ODM requirement at a later date.

malarzm reviewed (approved):
> which we may bump to 5.1.x

Please do, it's new config switch after all.

alcaeus commented:
> Please do, it's new config switch after all.

Will do once we have a 5.1.x branch - release automation doesn't handle pre-releases very gracefully, so once 5.0.0 is tagged everything should be in order.

malarzm commented:
Ah 5.0.0 is still in RC phase 🙈

GromNaN commented on tests/DependencyInjection/DoctrineMongoDBExtensionTest.php:
```diff
@@ -369,4 +372,70 @@ public function testControllerResolver(): void
+
+    public function testTransactionalFlushConfigurationWhenNotSupported(): void
+    {
+        if (method_exists(ODMConfiguration::class, 'setUseTransactionalFlush')) {
```
This is a weak test. If the method is renamed, the test is skipped while it should fail.

I think it would be more robust to use `Composer\InstalledVersions`:

```php
Composer\InstalledVersions::satisfies(new VersionParser, 'doctrine/mongodb-odm', '>=2.7@dev')
```

alcaeus replied:
Updated. I wasn't even aware this existed. This is simply lovely :)

GromNaN replied:
I see [3 skipped tests](https://github.com/doctrine/DoctrineMongoDBBundle/actions/runs/7584953100/job/20659752583?pr=814#step:9:15), is the version contraint correct?

GromNaN reviewed (approved):
A suggested improvement, but SGTM as this.

alcaeus committed:
Use InstalledVersions to check for transaction support

alcaeus committed:
Fix phpcs violations
